From 92597e456aba95798cf2a9a4367e6157311d2011 Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Fri, 28 Oct 2011 14:22:47 -0200
Subject: [RHEL6 qemu-kvm PATCH 16/17] Revert "migration: simplify state assignmente"

RH-Reverts: df2e96a1110c60f439bdf6ab074694c33af71554
RH-Reverts-patchwork-id: 34426
Bugzilla-related: 669581
Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 migration.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/migration.c b/migration.c
index d40005d..a62dcb7 100644
--- a/migration.c
+++ b/migration.c
@@ -391,6 +391,7 @@ void migrate_fd_put_ready(void *opaque)
 
     DPRINTF("iterate\n");
     if (qemu_savevm_state_iterate(s->mon, s->file) == 1) {
+        int state;
         int old_vm_running = runstate_is_running();
 
         DPRINTF("done iterating\n");
@@ -402,8 +403,11 @@ void migrate_fd_put_ready(void *opaque)
             if (old_vm_running) {
                 vm_start();
             }
-            s->state = MIG_STATE_ERROR;
+            state = MIG_STATE_ERROR;
+        } else {
+            state = MIG_STATE_COMPLETED;
         }
+        s->state = state;
 	STOP_MIGRATION_CLOCK();
 	DPRINTF("ended after %lu milliseconds\n", stop);
 
@@ -411,10 +415,8 @@ void migrate_fd_put_ready(void *opaque)
             if (old_vm_running) {
                 vm_start();
             }
-            s->state = MIG_STATE_ERROR;
         }
-        if (s->state == MIG_STATE_ACTIVE) {
-            s->state = MIG_STATE_COMPLETED;
+        if (state == MIG_STATE_COMPLETED) {
             runstate_set(RUN_STATE_POSTMIGRATE);
         }
         notifier_list_notify(&migration_state_notifiers);
-- 
1.7.3.2

